src/visualization/components/Player.tsx

```diff
@@ -87,45 +121,252 @@ class Player extends React.Component<any, any> {
     }
 
     playImpression(index) {
-        this.props.selectImpression(this.props.playlist[index]);
+        this.props.selectImpression(this.props.session[index]);
+    }
+
+    uploadAndShare() {
+        var state = { "showText": true, "showImages": true, "recording": false, "saving": false, "uploaded": false, "uploadedUrl": "" };
+        var thisRef = this;
+        thisRef.openShareWindow();
+
+        // Read from default storage
+        chrome.storage.local.get({ clarity: state }, function (items) {
+            state = items.clarity;
+
+            if (!state.uploaded) {
+                
+                chrome.runtime.sendMessage({ fetch: true }, function (response) {
+                    var rawData = response;
+                    var logJSON = JSON.stringify({ response });
+                    var len = 250000;
+                    var numChunks = Math.ceil(logJSON.length / len),
+                        splitLogs = new Array(numChunks);
+
+                    for (var i = 0, o = 0; i < numChunks; ++i, o += len) {
+                        splitLogs[i] = logJSON.substr(o, len);
+                    }
+                            
+                    var splitSize = splitLogs.length;
+                    
+                    $.ajax({
+                        type: 'GET',
+                        url: thisRef.extensionCreateBlobAPIEndpoint,
```

Comment:
>extensionCreateBlobAPIEndpoint [](start = 37, length = 30)

does it make sense for the client to call a create blob api? Should the client just upload bits and the server can decide whether it needs to create a new blob or not?